pdns/dnsdist.cc

```diff
@@ -1947,6 +1947,9 @@ static void* healthChecksThread()
 
     auto states = g_dstates.getLocal(); // this points to the actual shared_ptrs!
     for(auto& dss : *states) {
+      if(++dss->lastCheck < dss->checkInterval)
```

Comment:
I'm not sure I agree, the current code is easy to understand, while the one you suggest would require a bit more logic to 
keep track of the next health check regardless of the backend (to know for how long we need to sleep) AND the next health check per backend. It might also lead to unexpected surprises when a backend has just been added, for example if the new backend has to be checked every second but the existing ones were only checked every 60s, we might be in the middle of a 60s sleep..

Reply:
Oh right, there is one loop for multiple backends, in that case this is easier/better indeed.

Reply:
Thought about changing that sleep in two ways.
1) make checkInterval global for all servers so that sleep(1) would become sleep(checkInterval). But I assume most people (including me) want different intervals for the servers.
2) remove the sleep(1) and create logic to always wait for the next coming check. But that would add much more code as rgacogne said correctly. Always waking up after 1 second is not a pretty solution but I guess the code/value ratio makes this "ok" :)
I think there is room to improve server checking in the future. So maybe a overhaul might be the right choice later on.

Reply:
Ye I agree, this is the best solution without completely overhauling this.